junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/execution/ExtensionValuesStore.java

```diff
@@ -110,7 +110,16 @@ Object remove(Namespace namespace, Object key) {
 
 	private StoredValue getStoredValue(Namespace namespace, Object key) {
 		CompositeKey compositeKey = new CompositeKey(namespace, key);
-		return this.storedValues.get(compositeKey);
+
+		/*
+		 * This method can be called from another ExtensionValuesStore instance,
+		 * so we must synchronized on the monitor.
+		 *
+		 * TODO: Consider rewriting with ConcurrentHashMap.
```

Comment:
> I actually do. I found this bug while rewriting the class to use fine-grained locks. I wanted to send out the fix first.

So, are you saying we currently have a bug that would be fixed if we switched to using `ConcurrentHashMap` as the implementation for `ExtensionValuesStore.storedValues`?